test/integration/api/services_controller_test.rb

```diff
@@ -82,42 +82,39 @@ def update_params
     end
   end
 
-  class ActAsProduct < Api::ServicesControllerTest
+  class BackendApiCreation < Api::ServicesControllerTest
     def setup
       super
       Logic::RollingUpdates.stubs(enabled?: true)
       Account.any_instance.stubs(:provider_can_use?).returns(true)
       @provider.settings.allow_multiple_services!
     end
 
-    test 'should create a new Backend API if none was selected' do
+    test 'should not create the default Backend API if API as Product is enabled' do
       Account.any_instance.stubs(:provider_can_use?).with(:api_as_product).returns(true).at_least_once
 
-      assert_change of: -> { BackendApi } do
+      assert_change of: -> { BackendApi.count }, by: 0 do
```

Comment:
`assert_no_change`

Reply:
I think you should test for when the RU is disabled as well!

Reply:
there is one already

```
    test 'should create the default Backend API if API as Product is disabled' do
      Account.any_instance.stubs(:provider_can_use?).with(:api_as_product).returns(false).at_least_once
```